quantum/debounce/sym_defer_pr.c

```diff
@@ -21,51 +21,75 @@ DEBOUNCE milliseconds have elapsed since the last change.
 #include "timer.h"
 #include <stdlib.h>
 
+#ifdef PROTOCOL_CHIBIOS
+#    if CH_CFG_USE_MEMCORE == FALSE
+#        error ChibiOS is configured without a memory allocator. Your keyboard may have set `#define CH_CFG_USE_MEMCORE FALSE`, which is incompatible with this debounce algorithm.
+#    endif
+#endif
+
 #ifndef DEBOUNCE
 #    define DEBOUNCE 5
 #endif
 
-static uint16_t last_time;
+// Maximum debounce: 255ms
+#if DEBOUNCE > UINT8_MAX
+#    undef DEBOUNCE
+#    define DEBOUNCE UINT8_MAX
+#endif
+
+typedef uint8_t debounce_counter_t;
+
+#if DEBOUNCE > 0
+static fast_timer_t last_time;
 // [row] milliseconds until key's state is considered debounced.
-static uint8_t* countdowns;
+static debounce_counter_t* debounce_counters;
 // [row]
 static matrix_row_t* last_raw;
 
+#    define DEBOUNCE_ELAPSED 0
+
 void debounce_init(uint8_t num_rows) {
-    countdowns = (uint8_t*)calloc(num_rows, sizeof(uint8_t));
-    last_raw   = (matrix_row_t*)calloc(num_rows, sizeof(matrix_row_t));
+    const size_t debounce_counters_size = num_rows * sizeof(debounce_counter_t);
+    debounce_counters                   = (debounce_counter_t*)malloc(debounce_counters_size);
```

Comment:
Inconsistent formatting of pointer types between changes introduced in this file (`debounce_counter_t*`) and in `quantum/debounce/sym_defer_pk.c` (`debounce_counter_t *`). [Please run `clang-format`](https://docs.qmk.fm/#/coding_conventions_c?id=auto-formatting-with-clang-format) to lint your files.

This does mean that, while some of my suggestions can be applied directly on GitHub, they will need more attention, as I have kept your formatting in each instance.